src/kafka-util/src/bin/kgen.rs

```diff
@@ -107,33 +131,28 @@ impl<'a> RandomAvroGenerator<'a> {
                 scale,
                 fixed_size: _,
             } => {
-                let f = self.decimals.get_mut(&p).unwrap();
-                let unscaled = f();
+                let unscaled = self.decimals.get_mut(&p).unwrap()(rng);
                 Value::Decimal(DecimalValue {
                     unscaled,
                     precision: *precision,
                     scale: *scale,
                 })
             }
             SchemaPiece::Bytes => {
-                let f = self.bytes.get_mut(&p).unwrap();
-                let mut val = vec![];
-                f(&mut val);
```

Comment:
This is strange; I don't remember at all why I had these take the output vector as a function argument rather than just returning `Vec<u8>`. Any insight?

Reply:
Not really! I can only assume you wrote the signature of the generator functions to re-use buffers before you realized that the Avro API made that infeasible.

Reply:
That sounds plausible.